qlib/Swagger.qm

```diff
@@ -0,0 +1,3666 @@
+        OperationObject op = pio.getOperation(method, path);
+
+        hash<string, bool> mime_types;
+        op.validateResponse(method, path, pio, code, response_body, \mime_types);
+
+        hash<HttpResponseInfo> rv((
+            "code": code,
+            "hdr": headers,
+        ));
+
+        if (exists response_body) {
+            if (!exists content_types)
+                content_types = MimeContentTypes;
+            foreach string content in (content_types) {
+                if (!mime_types{content})
+                    continue;
+                if (content == MimeTypeMultipartFormData) {
+                    if (response_body.typeCode() != NT_HASH)
+                        throw "SERIALIZATION-ERROR", sprintf("response to %s %s: cannot serialize type %y to MIME type %y; expecing \"hash\"", method.upr(), path, response_body.type(), content);
+                    hash<string, hash<FormDataMessageInfo>> parts = cast<hash<string, hash<FormDataMessageInfo>>>(map {$1.key: cast<hash<FormDataMessageInfo>>((
+                        "name": $1.key,
+                        "filename": $1.key,
+                        "hdr": ("Content-Type": MimeTypeText),
+                        "body": $1.value,
+                    ))}, response_body.pairIterator());
+
+                    hash<MessageInfo> mh = MultiPartFormDataMessage::makeMessage(parts).getMsgAndHeaders();
+                    return rv += (
+                        "body": mh.body,
+                        "hdr": rv.hdr + ("Content-Type": mh.hdr."Content-Type"),
+                    );
+                }
+                if (!MimeDataTypes{content})
+                    continue;
+                return rv += (
+                    "body": compact_serialization ? MimeDataTypes{content}.serialize(response_body) : MimeDataTypes{content}.serialize_verbose(response_body),
+                    "hdr": rv.hdr + ("Content-Type": content),
+                );
+            }
+            if (content_types == MimeContentTypes)
+                throw "ACCEPT-ERROR", sprintf("response to %s %s: message body cannot be serialized; available MIME type(s) for this call: %y; available serialization modules: %y", method.upr(), path, mime_types, SerializationModules), ("types": mime_types);
+            else
+                throw "ACCEPT-ERROR", sprintf("response to %s %s: message body cannot be serialized; requested MIME type(s): %y; available MIME types for this call: %y; available serialization modules: %y", method.upr(), path, content_types, mime_types, SerializationModules), ("types": mime_types);
+        }
+
+        return rv;
+    }
+}
+
+#! The object provides metadata about the API. The metadata can be used by the clients if needed, and can be presented in the %Swagger-UI for convenience.
+public class InfoObject inherits ObjectBase {
+    public {
+        #! Required. The title of the application.
+        string title;
+
+        #! A short descripti
```

Comment:
There is no `float` swagger type, only `number` and that is equivalent to Qore's `float`... For specifying more precisely, there's the `format` field.

Reply:
fixed